virttest/test_setup.py

```diff
@@ -1284,20 +1286,20 @@ def get_same_group_devs(self, pci_id):
             pci_ids = out.split()
         return pci_ids
 
-    def get_pf_mlx(self):
+    def get_pf_ifname(self, pci_pf):
         """
-        Get the PF devices for mlx5_core driver
+        Get the ifname of a PF device
+
+        :params pci_pf: Pci id of the pf device
         """
-        # This function returns the network device associated with
-        # mellanox cx4 card
-        cmd = "lshw -c network -businfo | grep '%s' | awk '{print $2}'" % self.pf_filter_re
-        PF_devices = process.system_output(
-            cmd, shell=True, timeout=60).splitlines()
-        return (PF_devices)
+        # This function returns the interface name associated with the pf
+        cmd = "ls /sys/bus/pci/devices/%s/net/" % pci_pf
+        ifname = process.system_output(cmd, shell=True, timeout=60)
+        return ifname
 
     def get_pf_ids(self):
         """
-        Get the PF devices on x86 host
+        Get the id of PF devices
```

Comment:
even this function can be removed with `get_pci_id_using_filter()` from utils_misc, these will make your changes more good.

Reply:
I'm afraid it can't be replaced with `get_pci_id_using_filter()`.  `get_pci_id_using_filter()` can only filter the device by a filter regex. We need to exclude 'Virtual Function' here by `grep -v` as L1304. Thanks!